source/leuven/services/openondemand.rst

```diff
@@ -191,136 +226,244 @@ of using any of the other partitions. There are however some limitations on the
+
+Once you launch a JupyterLab session, a default kernel called ``Python 3 (ipykernel)`` is already available in your session.
+This kernel, in addition to the Python standard library, would enable using extra packages from
+``SciPy-bundle`` and/or ``matplotlib``, if you already selected them.
+
+If the "standard" environment explained above does not provide all packages that you need, then
+it is recommended to manage your custom-made R or Python environments using Conda.
+You should :ref:`install Miniconda <install_miniconda_python>` if you have not installed it yet.
 
-To create any other kernels, first create a  :ref:`Python <create_python_conda_env>` or 
-:ref:`R <create_r_conda_env>` conda environment. The second step consists out of effectively
-creating the kernel in your ``$VSC_HOME/.local`` folder, as Jupyter will look for kernels in this location. The following commands should be excecuted from a shell, 
-and only need to be done once for the set-up of each new kernel. This starts with activating your conda
-environment::
+To create any other kernel, first create a  :ref:`Python <create_python_conda_env>` or
+:ref:`R <create_r_conda_env>` Conda environment. The second step consists out of effectively
+creating the kernel in your ``$VSC_HOME/.local`` folder, as Jupyter will look for kernels in this location.
+The following commands should be excecuted from a shell (e.g. using 'Login Server Shell Access'), and only need
+to be done once for the set-up of each new kernel.
+If you already have an existing Python kernel, but your JupyterLab session freezes/craches when choosing your
+old kernel, you also need to repeat the following steps only once.
+This starts with activating your Conda environment::
 
       source activate <env_name>
 
-For Python you will need the ``ipykernel`` package installed in your conda environment. Then you create the kernel as follows::
+For Python you will need the ``ipykernel`` package installed in your Conda environment::
 
-      python -m ipykernel install --user --name '<env_name>' --display-name '<kernel_name>'
+      conda install ipykernel
+
+Then you create the kernel as follows::
+
+      python -m ipykernel install --user --env PYTHONPATH "" --name '<env_name>' --display-name '<kernel_name>'
 
 For R, you need both the ``jupyter_client`` and the ``irkernel`` package installed. With the following command you can create the kernel::
 
       Rscript -e 'IRkernel::installspec(name="<env_name>", displayname="<kernel_name>")'
 
-Once the kernel is created, you will see it in the 'Launcher' menu. You can now start working in your own customized environment.
+Once the kernel is (re)created, you will see it in the 'Launcher' menu. You can now start working in your own customized environment.
+Note that all user kernels are stored by default in ``${VSC_HOME}/.local/share/jupyter/kernels``, but we strongly advice you
```

Comment:
(same remark here as above, about ``$XDG_DATA_HOME``)

Reply:
Sure. Let me know what you think here.

Reply:
(I think this is now further discussed in a new thread)